bot/cogs/source.py

```diff
@@ -0,0 +1,80 @@
+import inspect
+import logging
+
+import discord
+from discord.ext import commands
+
+from bot import constants
+from bot.decorators import in_channel
+
+logger = logging.getLogger(__name__)
+
+
+class Source(commands.Cog):
+    """Send source code url for a specific command."""
+
+    def __init__(self, bot: commands.Bot) -> None:
+        self.bot = bot
+
+    @in_channel(constants.Channels.bot)
+    @commands.command(name="source")
+    async def command_source(self, ctx: commands.Context, command_name: str = None, sub_command: str = None) -> None:
+        """View the source of a command."""
+        if command_name is None:
+            await ctx.send("> https://github.com/python-discord/bot")
+            return
+
+        if sub_command:
+            await ctx.send(sub_command)
+            command = self.bot.get_command(f"{command_name} {sub_command}")
+        else:
+            command = self.bot.get_command(command_name)
+
+        if command is None:
+            await ctx.send("No such command found.")
+            return
+
+        url = self.get_command_url(command)
+
+        prefix = constants.Bot.prefix
+
+        embed = discord.Embed(colour=discord.Colour.red())
+        embed.title = "Command Source"
+        embed.description = f"**{command.name.capitalize()}**\n"
+        embed.description += f"`{prefix}{command.name}`\n\n {url}"
+
+        await ctx.send(embed=embed)
+
+    @staticmethod
+    def get_command_url(command: commands.Command) -> str:
+        """Make up the url for the source of the command."""
+        # Get the source code
+        src_code_object = command.callback.__code__
+        file_path = src_code_object.co_filename[5:]
```

Comment:
Rapptz's implementation of getting the location was like this: `file_name = src_code_object.co_filename` and `file_path = os.path.relpath(file_name).replace('\\', '/')`. I think, should work; will do some tests in a bit.